### ISSUE TITLE
$model->getPrimaryKey() returns wrong value when yii\log\DbTarget is used for loging

### ISSUE THREAD

**Issue by TomasMolnar** [yiisoft/yii2#12171]
###setup
Model table has "id" column set as primary key, autoincrement

### What steps will reproduce the problem?
$model = new Model();
$model->name = 'model name';
$model->save();

//this should return the id of newly inserted row to Model table, 
//but returns the id of row inserted  to Log table instead
echo $model->getPrimaryKey();

### What is the expected result?
expected is a newly assigned id according to AutoIncremen PK

### What do you get instead?
returns the id of row inserted  to Log table instead

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      |  2.0.9
| PHP version      |  7.0.5
| DB version    | MariaDB 5.5
| Operating system |  Linux apache-data-it11-113 3.14.27-031427-generic #201412170558 SMP Wed Dec 17 11:00:01 UTC 2014 x86_64

config/web.php: 
 '```
log' => [
            'traceLevel' => 0,//YII_DEBUG ? 3 : 0,
            'flushInterval' => YII_DEBUG ? 1 : 100,
            'targets' => [
            	[
            		'class' => 'yii\log\DbTarget',
            		'exportInterval' => YII_DEBUG ? 1 : 10,
            		'levels' => ['error', 'warning', 'profile' /*, 'info', 'trace', */],
            		'logVars' => [],	
            		'categories' => [
            			'application',
            			'yii\db\*',
            			'yii\web\HttpException:*',	
            		],
            		'except' => [
            			/*	
            			'yii\web\HttpException:404',
            			'yii\web\HttpException:429',
            			*/
            		],
            		'prefix' => function ($message) {
            			$user = Yii::$app->has('user', true) ? Yii::$app->get('user') : null;
            			$userID = $user ? $user->getId(false) : '-';
            			return json_encode(['userid' => $userID]);
            		},
            	]
            ],
        ]
```

**Comment by cebe** [comment]
This is a limitation of PDO: http://php.net/manual/en/pdo.lastinsertid.php not sure how to work around this. One problem could be the exportInterval of 1, which means that the message is written to the database immediately. If you set it to a higher value you can avoid this problem to some extent.

**Comment by TomasMolnar** [comment]
thanks @cebe for verifying. However I think, that setting a higher exportInterval might cause even more weird behavior as this error would arise randomly when the log stack is flushed to DB. Would it be possible to add some log restriction to core to avoid such setting? Or at least echo some warning that this setting causes problems when the **$model->id** is used as **FK** for saving subsequent related models?

**Comment by cebe** [comment]
to solve the issue completely the best way would be to exclude database queries from the logging of the database target.

**Comment by klimov-paul** [comment]
Note that databases which use `INSERT ... RETURNING` such as Oracle and PostgreSQL are not affected by this issue.

**Comment by Mak-Di** [comment]
What do you think about moving [getLastInsertID](https://github.com/yiisoft/yii2/blob/master/framework/db/Schema.php#L458) before [Yii::endProfile($token, \__METHOD__);](https://github.com/yiisoft/yii2/blob/master/framework/db/Command.php#L846) ?

**Comment by samdark** [comment]
Using another connection makes sense as a workaround. Does moving getLastInsertID solve issue for all possible cases?

**Comment by Mak-Di** [comment]
> Does moving getLastInsertID solve issue for all possible cases?

No, definitely not. It was not a very good suggestion.

**Comment by AnatolyRugalev** [comment]
> Using another connection makes sense as a workaround.

So, is this about target configuration, right? Or may be we can make a workaround inside of `DbTarget` which will instantiate new connection?

**Comment by cebe** [comment]
@AnatolyRugalev yes, if you configure `$db` property of the DB target to not use 'db' application component but a separate DB connection.

**Comment by cebe** [comment]
I think the only way to solve this would be to document it so users can adjust the configuration to avoid the problem.

**Comment by klimov-paul** [comment]
At the `Schema::insert()` we are passing `$sequenceName` to the `PDO::lastInsertId()`:
https://github.com/yiisoft/yii2/blob/master/framework/db/Schema.php#L456

However for MySQL it is always empty. Perhaps if we could fetch full sequece name for the table and pass it here there will be such confusion. Although I am unsure if it will work or if fetching sequence name from MySQL is possible.

**Comment by cebe** [comment]
As far as I know, MySQL PDO does not have a concept like sequence name. last insert id is maintained per connection, not per table or db: 

- https://dev.mysql.com/doc/refman/5.7/en/getting-unique-id.html
- https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_last-insert-id
